src/source.cc

```diff
@@ -2120,7 +2194,25 @@ bool Source::View::on_button_press_event(GdkEventButton *event) {
     get_buffer()->select_range(start, end);
     return true;
   }
-  
+  if((event->type == GDK_BUTTON_PRESS) && (event->button == 3)){
+    if(!get_buffer()->get_has_selection()){
+      int x,y;
+      window_to_buffer_coords(Gtk::TextWindowType::TEXT_WINDOW_WIDGET,event->x,event->y,x,y);
```

Comment:
The x coordinate was somewhat off target. Fixed by using `Gtk::TextWindowType::TEXT_WINDOW_TEXT` instead of `Gtk::TextWindowType::TEXT_WINDOW_WIDGET`.